core/src/transport.rs

```diff
@@ -221,149 +221,252 @@ pub trait Transport {
     }
 }
 
-/// Event produced by [`Transport::Listener`]s.
+/// The ID of a single listener.
+#[derive(Copy, Clone, Debug, PartialEq, Eq, Hash, PartialOrd, Ord)]
+pub struct ListenerId(u64);
+
+impl ListenerId {
+    /// Creates a `ListenerId` from a non-negative integer.
+    pub fn new(id: u64) -> Self {
+        Self(id)
+    }
+}
+
+impl std::ops::Add<u64> for ListenerId {
+    type Output = Self;
+
+    fn add(self, other: u64) -> Self {
+        Self(self.0 + other)
+    }
+}
```

Comment:
If we end up doing the namespace thing, I believe we should make this type completely opaque and remove any reference to `u64` from the public API.